### ISSUE TITLE
Support source file reloads in GHCi session

### ISSUE THREAD

**Issue by gdeest** [tweag/rules_haskell#173]
This is related to both #159 and #82 .

Current prototype ( #159 ) gives a mean to load a GHCi prompt with a set of compiled libraries for quick experimentation. However, this is insufficient for interactive development, for example with `ghcid`. For this to work, we need to load source modules, not their compiled form.

ATM, I can work around this issue by only specifying the dependencies of my library and adding source modules manually to the GHCI command this way, but this is not satisfactory:

```
ghcid --command="./bazel-bin/mylib-repl -isrc src/My/Module.hs ..."
```

Ideally, I'd like to write the following:

```
haskell_repl(
  name = "mylib-repl",
  deps = [ ... ],
  srcs = [":mylib"],
  prebuilt_dependencies = [ ... ]
)
```

and have all `:mylib`'s modules loaded from source into GHCi.

**Comment by gdeest** [comment]
In fact, this problem is not solely about integration with `ghcid` and similar tools. I can't really use `haskell_repl` for interactive development.

Say I have the following `BUILD` file:

```
haskell_library(
    name = "mylib",
    src_strip_prefix = "src",
    srcs = ["src/Lib.hs"],
)

haskell_repl(
  name = "mylib-repl",
  deps = ["mylib"],
)
```

Assuming the `mylib` library builds successfully, I can build the REPL with `bazel build :mylib-repl` and then fire ghci with `./bazel-bin/mylib-repl`. However, if I modify `src/Lib.hs` and hit `:r`, it will not reload the *new* version of module `Lib`, as it does with `stack ghci` and `cabal repl`. Instead, it will simply reload the old version.

What I would like is a way to get a REPL that loads *source versions* of my library's modules, to get quick feedback during development, not the compiled form.

**Comment by mboes** [comment]
Retitling ticket. Old ticket title made it look like a dupe of #82.

**Comment by mrkkrp** [comment]
Reloading per se is supported with what we merged in #159, integration with ghcid could be improved though.

**Comment by mrkkrp** [comment]
We talked with @gdeest and it looks this particular issue may now be closed.
